### DISCUSSION TITLE
Question on Running sharpy with atmospheric turbulence

### DISCUSSION THREAD

**Discussion by xxyao-95** [ImperialCollegeLondon/sharpy#142] [Q&A] [answered]
Hi, 

I have generated atmospheric turbulence using Von-Karman turbulence model from a gaussian white noise input. I want to apply this turbulence to a sharpy simulation. I would like to ask that which sharpy setting is the best to use in this situation.

The atmospheric turbulence velocity file I generated is in the form of the following:

Time(s)  delta_u(m/s)  delta_v(m/s)  delta_w(m/s)
0.01            12                     13               14
.......

The simulation that I want to run is a time domain simulation with goland wing geometry. Here is the list of solvers that I am using:
ws2.config['SHARPy']['flow'] = [
    'BeamLoader',
 'AerogridLoader',
 'StaticCoupled',
 'DynamicCoupled',
 'AerogridPlot',
 'BeamPlot',
 'SaveData'
]
I am using "NonLinearDynamicPrescribedStep" for structural solver and "StepUvlm" for aero solver.

I am a bit confused with which setting to use to apply the atmospheic turbulence generated into the simulation. I am looking at 3 possible settings and I am not sure which one is the most suitable:

The first is using ``gust_shape = 'time varying'`` in `GustVelocityField` setting and read from the file containing the velocities.
The second is using the "TurbVelocityField" setting.
The third is creating a .dyn.h5 file containing the velocity.

Could you let me know which one is the most appropriate?

**Comment by ArturoMS13** [comment, marked as answer]
Dear xxyao-95,

In your case, the most suitable option is using `gust_shape = 'time varying'` (https://ic-sharpy.readthedocs.io/en/master/includes/generators/gustvelocityfield/time_varying.html). Include the route to your file in the `file` setting.

Just for the record, `TurbVelocityField` can be used if the wind has been generated with TurbSim (https://www.nrel.gov/wind/nwtc/turbsim.html) into a bts file. Moreover, the `.dyn.h5` file is used to prescribe solid velocities or applied forces.

I hope this helps.

**Reply by xxyao-95** [reply]
Hi ArturoMS13

Thank you so much! I understand what to do now.
